packages/components/src/combobox-control/stories/index.js

```diff
@@ -287,3 +287,76 @@ _default.args = {
 	__next36pxDefaultSize: false,
 	allowReset: false,
 };
+
+const authors = [
+	{
+		name: 'Hermann P. Schnitzel',
+		age: 45,
+		country: 'Germany',
+	},
+	{
+		name: 'Shequondolisa Bivouac',
+		age: 43,
+		country: 'France',
+	},
+	{
+		name: 'Bodrum Salvador',
+		age: 42,
+		country: 'Spain',
+	},
+	{
+		name: 'Parsley Montana',
+		age: 48,
+		country: 'Germany',
+	},
+	{
+		name: 'Cabbage New York',
+		age: 44,
+		country: 'France',
+	},
+	{
+		name: 'Jake Weary',
+		age: 41,
+		country: 'United Kingdom',
+	},
+];
+
+const authorOptions = authors.map( ( { name, ...details } ) => ( {
+	value: name,
+	label: name,
+	...details,
+} ) );
+
+export const WithRenderOption = ( args ) => {
+	const [ value, setValue ] = useState( '' );
+
+	return (
+		<>
+			<ComboboxControl
+				{ ...args }
+				value={ value }
+				onChange={ setValue }
+				label="Select an author"
+				options={ authorOptions }
+				__experimentalRenderItem={ ( { item } ) => {
+					const { label, age, country } = item;
+					return (
+						<div>
+							<div style={ { marginBottom: '0.2rem' } }>
+								{ label }
+							</div>
+							<small>
+								Age: { age }, Country: { country }
+							</small>
+						</div>
+					);
+				} }
+			/>
+			<p>Selected author: { value }</p>
+		</>
+	);
+};
+
+WithRenderOption.args = {
+	allowReset: false,
+};
```

Comment:
Let's also add this quick story to `form-token-field/stories/index.tsx`:

```tsx
/**
 * The rendered output of each suggestion can be customized by passing a
 * render function to the `__experimentalRenderItem` prop. (This is still an experimental feature
 * and is subject to change.)
 */
export const WithCustomRenderItem: ComponentStory< typeof FormTokenField > =
	DefaultTemplate.bind( {} );
WithCustomRenderItem.args = {
	...Default.args,
	__experimentalRenderItem: ( { item } ) => (
		<div>{ `${ item } — a nice place to visit` }</div>
	),
};
```

Reply:
Added, thank you.
<img width="805" alt="Screenshot 2022-08-03 at 5 51 12 PM" src="https://user-images.githubusercontent.com/18226415/182606380-6c492e2c-f2d6-4272-9110-7e220cf568fb.png">